### DISCUSSION TITLE
What is the difference between accelerometer, gyroscope and magnetometer?

### DISCUSSION THREAD

**Discussion by eagl1** [jrowberg/i2cdevlib#609] [Q&A]

**Comment by eagl1** [comment]
I've been watching YouTube videos about all these modules but now I got confused because they seem to do the same job which is to detect tilt effects! What is the actual difference between these modules?﻿

What I know is the follows:
1. Gyroscope is for measuring the angular speed or rate, if that's correct, then how could I get the benefit from the raw data of this module?
2. Accelerometer is for measuring the acceleration in any of the 3 dimensions. Which means that the values are related to time. But how it's really working what's the max rate of the acceleration of this module? Can it measure high accelerations of something like an aircraft taking off?
3. Magnetometer for measuring the angles of x,y to get the heading then z for how I'm near of far from the earth surface.

How is my understanding for the modules?

**Comment by ErikBob** [comment]
I don't really think this question belongs here as it has nothing to do with the library.

1. The benefit of measuring angular momentum depends on your application.
2. On the least sensitive setting you can measure up to 16g, so an aircraft shouldn't be a problem
3. x,y and z are all needed to get the heading as it is measured relative to the body and not the inertial frame. If you tilt the sensor your heading will have an z-component.

**Comment by eagl1** [comment]
> I don't really think this question belongs here as it has nothing to do with the library.
> 
> 1. The benefit of measuring angular momentum depends on your application.
> 2. On the least sensitive setting you can measure up to 16g, so an aircraft shouldn't be a problem
> 3. x,y and z are all needed to get the heading as it is measured relative to the body and not the inertial frame. If you tilt the sensor your heading will have an z-component.

Thanks for the answer, but I want to learn some aspects about this module. It's just I need more revisions to understand all the three sensors, and to think which one is enough for a quadcopter. Are they different? Do they have some similarities? Or each one has a very different function and more suitable for specific applications.
